src/openrct2-ui/windows/RideList.cpp

```diff
@@ -813,21 +879,25 @@ namespace OpenRCT2::Ui::Windows
         template<typename TSortPred>
         void SortListByPredicate(const TSortPred& pred)
         {
-            std::sort(_rideList.begin(), _rideList.end(), [&pred](const auto& lhs, const auto& rhs) {
+            bool desc = _windowListSortDescending;
+            std::sort(_rideList.begin(), _rideList.end(), [&pred, desc](const auto& lhs, const auto& rhs) {
                 const Ride* rideLhs = GetRide(lhs.Id);
                 const Ride* rideRhs = GetRide(rhs.Id);
                 if (rideLhs == nullptr || rideRhs == nullptr)
                 {
                     return rideLhs != nullptr;
                 }
-                return !pred(*rideLhs, *rideRhs);
+                auto result = !pred(*rideLhs, *rideRhs);
+                return desc ? !result : result;
```

Comment:
This looks very innocuous to me, but somehow this now crashes in 'descending' mode.

Reply:
Turns out changing `auto` to `bool` fixes the issue.